File: linear/linked_list/singly_linked_list/SinglyLinkedList.java
class Node {
    int value;
    Node nextNode;

    public Node(int value) {
        this.value = value;
        this.nextNode = null;
    }
}

public class SinglyLinkedList {
    private Node head;

    public SinglyLinkedList() {
        this.head = null;
    }

    public Node getHead() {
        return this.head;
    }

    public int getSize() {
        int n = 0;

        if (this.head == null) {
            return n;
        }

        Node current = this.head;
        while (current != null) {
            n ++;
            current = current.nextNode;
        }

        return n;
    }
}

class Test {
    public static void main(String[] args) {
        SinglyLinkedList ll = new SinglyLinkedList();
        System.out.println(ll.getSize());
    }
}

File: linear/linked_list/singly_linked_list/singly_linked_list.ts
interface Node {
    value: number;
    nextnode: Node | null;
}

class SinglyLinkedList {
    private head: Node | null;

    constructor(head: Node | null = null) {
        this.head = head;
    }

    getHead(): Node | null {
        return this.head;
    }

    getSize(): number {
        let n: number = 0;
        let current: Node | null = this.head
        
        while (current != null) {
            n ++;
            current = current.nextnode;
        }

        return n;
    }

    
}

export {
    Node,
    SinglyLinkedList
}